unity/Assets/Maroon/reusablePrefabs/NewRuler/Scripts/RulerLogic.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using Maroon.Physics;
+using Maroon.Physics.CoordinateSystem;
+using UnityEngine;
+using Maroon.GlobalEntities;
+
+namespace Assets.Maroon.reusablePrefabs.NewRuler.Scripts
+{
+    public class RulerLogic : MonoBehaviour, IResetWholeObject
+    {
+        public GameObject RulerStart;
+        public GameObject RulerEnd;
+        public LineRenderer RulerLine;
+        private QuantityFloat _pinDistance = 0;
+
+        public QuantityFloat PinDistance 
+        { 
+            get => _pinDistance; 
+            set => _pinDistance = value;
+        }
+        private Vector3 _resetPosition; 
+
+        void Start()
+        {
+            _resetPosition = transform.position;
+
+            if (RulerLine.positionCount < 2)
+            {
+                RulerLine.SetPositions(new []{RulerStart.transform.position, RulerEnd.transform.position});
+            }
+
+            PinDistance.Value = 0f;
+        }
+
+        // Change this to a event system??
+        void Update()
+        {
+            if (RulerStart.activeSelf && RulerEnd.activeSelf)
+            {
+                if (!RulerLine.gameObject.activeSelf)
+                    RulerLine.gameObject.SetActive(true);
+
+                RulerLine.SetPosition(0, RulerStart.transform.position);
+                RulerLine.SetPosition(1, RulerEnd.transform.position);
+            }
+            else if (RulerLine.gameObject.activeSelf)
+            {
+                RulerLine.gameObject.SetActive(false);
+                PinDistance.Value = 0f;
+            }
+        }
+
+        public float CalculateDistance(Unit targetUnit = Unit.m)
+        {
+            var calcDist = CoordSystemHelper.Instance.GetSystemDistance(RulerStart.transform.position, RulerEnd.transform.position, targetUnit);
```

Comment:
maybe remove the GetSystemDistance method from the CoordSystemHelper and use just
Vector3.Distance with the SystemPosition from the GetSystemPosition method in this function